polaris/utils/progress_indicator.py

```diff
@@ -0,0 +1,26 @@
+from yaspin import yaspin
+from yaspin.spinners import Spinners
+
+
+class ProgressIndicator:
```

Comment:
**Suggestion**
It's fine as is, but it might be easier to use if it was written as a [context manager](https://realpython.com/python-with-statement/#creating-function-based-context-managers). Something like:
```python
@contextmanager
def progress_indicator(start_msg: str = 'In progress...', success_msg: str | None = None, error_msg: str | None = None):
    spinner = yaspin()
    spinner.spinner = Spinners.dots
    spinner.text = start_msg
    try:
        spinner.start()
        yield spinner
        spinner.color = "green"
        spinner.text = ""
        spinner.ok(f"✅ SUCCESS: \033[1m{success_msg}\033[0m\n")
    except Exception as err:
        spinner.color = "red"
        spinner.fail(f"💥 ERROR: {error_msg}")
        raise err
    finally:
        spinner.stop()
```

That would let you do this in the calling code:
```python
with progress_indicator("Uploading benchmark... ", "Your benchmark has been successfully uploaded to the Hub..."):
    ...do the work here...
```

Reply:
I'll give this a go, it looks cleaner 👍

Reply:
Some minor suggestions:

```python
@contextmanager
def progress_indicator(start_msg: str = 'In progress...', success_msg: str | None = None, error_msg: str | None = None):
    spinner = yaspin(Spinners.dots, text=start_msg)

    try:
        spinner.start()
        yield spinner
        spinner.text = ""
        spinner.green.ok(f"✅ SUCCESS: \033[1m{success_msg}\033[0m\n")
    except Exception as err:
        spinner.text = ""  # <-- Do we want to add this here too? 
        spinner.red.fail(f"💥 ERROR: {error_msg}")
        raise err
    finally:
        spinner.stop()
```

Reply:
Yes! Included in the latest commit.